source/fe/fe_poly_tensor.cc

```diff
@@ -84,66 +83,82 @@ namespace internal
                 const unsigned int nn = cell->neighbor_face_no(f);
 
                 if (nn < GeometryInfo<dim>::faces_per_cell / 2)
-                  for (unsigned int j = 0; j < dofs_per_face; ++j)
+                  for (unsigned int j = 0; j < fe.dofs_per_face; ++j)
                     {
-                      Assert(f * dofs_per_face + j < face_sign.size(),
+                      const unsigned int cell_j = fe.face_to_cell_index(j, f);
+                      // something in FiniteElement or FiniteElementData base
+                      // classes
+
+                      Assert(f * fe.dofs_per_face + j < face_sign.size(),
+                             ExcInternalError());
+                      Assert(mapping_type.size() == 1 ||
+                               cell_j < mapping_type.size(),
                              ExcInternalError());
 
                       // TODO: This is probably only going to work for those
                       // elements for which all dofs are face dofs
-                      face_sign[f * dofs_per_face + j] = -1.0;
+                      if (mapping_type.size() > 1 ?
+                            mapping_type[cell_j] == mapping_raviart_thomas :
+                            mapping_type[0] == mapping_raviart_thomas)
```

Comment:
It's not wrong to use the `?:` operator in an `if` clause, but it's often easier to read to write things out:
```
  if ((cond && expr1) || (!cond && expr2))
```

Reply:
Or here maybe:
```
  if ((mapping_type.size() > 1 ? mapping_type[cell_j] : mapping_type[0]) == mapping_raviart_thomas)
```
This probably expresses most clearly what you are doing.

Reply:
Yes I like that second option.